### PR TITLE
design-system: `Button` component implementation

### PR DESCRIPTION
**Pull request by manojVivek**

### CODE REVIEW

**Review by yomete — APPROVED** [review]
lgtm! 🚢🇮🇹

File: ui/packages/shared/components/src/Button/index.tsx [hunk 1]
```diff
@@ -38,58 +42,63 @@ export const BUTTON_VARIANT = {
...
+    );
+    const classesMerged = twMerge(classes);
 
-export default Button;
+    if (classes.length > classesMerged.length) {
+      const classesTokens = classes.split(' ');
+      const classesMergedTokens = classesMerged.split(' ');
+      console.warn(
+        'Button: Conflicting classes found in `className` prop, please use/create an appropriate variant instead. Conflicting classes:',
+        classesTokens.filter(token => !classesMergedTokens.includes(token))
```
[the remark this answers is not in the document]
Comment (manojVivek) [reply ui/packages/shared/components/src/Button/index.tsx:81]: Yes, but I'm still thinking if there is a way to make this logged only in development and not in prod.

[the remark this answers is not in the document]
Comment (monicawoj) [reply ui/packages/shared/components/src/Button/index.tsx:81]: I agree, if possible that would be better. Maybe we can just check if `process.env.NODE_ENV === "production"` and only warn in the case that it does not? 

This post has two interesting suggestions: https://dev.to/rajeshroyal/reactjs-disable-consolelog-in-production-and-staging-3l38

[the remark this answers is not in the document]
Comment (manojVivek) [reply ui/packages/shared/components/src/Button/index.tsx:81]: Hmm, was thinking about `NODE_ENV` too, but it's not set in some instances (eg. storybook), and we need to explicitly set it everywhere and add it to the `type` declaration for it to work smoothly. Will look into that. Thanks for the suggestion. 👍🏼

**Pull request merged**